packages/tss/src/provider.ts

```diff
@@ -0,0 +1,106 @@
+import { KeyType, Point, TorusServiceProviderArgs } from "@tkey/common-types";
+import { TorusServiceProvider } from "@tkey/service-provider-torus";
+import { PointHex } from "@toruslabs/rss-client";
+import TorusUtils from "@toruslabs/torus.js";
+
+import { getExtendedVerifierId } from "./util";
+
+export class TSSTorusServiceProvider extends TorusServiceProvider {
+  verifierName?: string;
+
+  verifierId?: string;
+
+  sssNodeDetails: {
+    serverEndpoints: string[];
+    serverPubKeys: PointHex[];
+    serverThreshold: number;
+  };
+
+  tssTorus: TorusUtils;
+
+  tssKeyType: KeyType;
+
+  /**
+   * Initializes this service provider instance. If `tssKeyType` is not
+   * provided, it defaults to the value of `keyType`.
+   */
+  constructor(args: TorusServiceProviderArgs & { enableOneKey?: boolean; tssKeyType?: KeyType }) {
+    super(args);
+    const { customAuthArgs, enableOneKey, tssKeyType } = args;
+    this.tssKeyType = tssKeyType || this.keyType;
+    this.tssTorus = new TorusUtils({
```

Comment:
This is for the case where `tssKeyType` is different than `keyType`. We still support this. It would be actually quite a bit of work to remove this separation now. Although the default is to not specific `tssKeyType` and in this case it will just default to `keyType`, as described above.

Reply:
if tssKeyType is optional and doesnt have much real usecase significance we should just remove it. We can add it later if any usecases arises.